app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsViewModel.kt

```diff
@@ -0,0 +1,226 @@
+            is EventDetailsEvent.AddPhoto -> addPhoto(event.uri)
+            EventDetailsEvent.ChangeIsGoingState -> changeIsGoingState()
+            EventDetailsEvent.ChangeMode -> changeMode()
+            EventDetailsEvent.CreateEvent -> createEvent()
+            EventDetailsEvent.DeleteEvent -> deleteEvent()
+            is EventDetailsEvent.FromDateTimeChanged -> changeFromDateTime(event.time)
+            is EventDetailsEvent.ReminderChanged -> changeReminder(event.reminderRange)
+            is EventDetailsEvent.ToDateTimeChanged -> changeToDateTime(event.time)
+            EventDetailsEvent.UpdateEvent -> updateAgendaItemEvent()
+            is EventDetailsEvent.DeletePhoto -> deletePhoto(event.key)
+        }
+    }
+
+    private fun deletePhoto(key: String) {
+        val deletedPhotoKeys = mutableListOf<String>()
+        deletedPhotoKeys.addAll(state.deletedPhotoKeys)
+        deletedPhotoKeys.add(key)
+        updateState(state.copy(deletedPhotoKeys = deletedPhotoKeys))
+    }
+
+    private fun updateAgendaItemEvent() {
+        viewModelScope.launch {
+            eventUseCases.updateEvent(
+                event = state.agendaItem,
+                deletedPhotoKeys = state.deletedPhotoKeys,
+                isGoing = state.isGoing
+            ).onSuccess {
+                channel.send(UiEventDetailsEvent.ShowInfoToast(it.countOfSkippedMessages.toString()))
+                channel.send(UiEventDetailsEvent.OnBackClick)
+            }.onFailure {
+                channel.send(UiEventDetailsEvent.ShowErrorMessage(R.string.default_event_details_error))
+            }
+        }
+    }
+
+    private fun changeIsGoingState() {
+        viewModelScope.launch {
+            eventUseCases.updateEvent(
+                event = state.agendaItem,
+                deletedPhotoKeys = state.deletedPhotoKeys,
+                isGoing = !state.isGoing
+            ).onSuccess {
+                channel.send(UiEventDetailsEvent.ShowInfoToast(it.countOfSkippedMessages.toString()))
+                channel.send(UiEventDetailsEvent.OnBackClick)
+            }.onFailure {
+                channel.send(UiEventDetailsEvent.ShowErrorMessage(R.string.default_event_details_error))
+            }
+        }
+    }
+
+    private fun changeReminder(reminderRange: ReminderRange) {
+        updateState(state.copy(reminderRange = reminderRange))
+        val agendaItem = state.agendaItem
+        val newRemindAt = agendaItem.time.minusMinutes(reminderRange.minutes)
+        updateEvent(
+            agendaItem.copy(
+                remindAt = newRemindAt,
+            )
+        )
+    }
+
+    private fun changeFromDateTime(time: LocalDateTime) {
+        updateEvent(state.agendaItem.copy(time = time))
+        changeReminder(state.reminderRange)
+    }
+
+    private fun changeToDateTime(time: LocalDateTime) {
+        updateEvent(state.agendaItem.copy(to = time))
+    }
+
+    private fun createEvent() {
```

Comment:
Update and create can call the same method, and then have the workmanager decide if it should call create or update